src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1703,7 +1703,10 @@ void Graph::EnforceInferencePrecision() {
         if (nodesToSkip.count(node) && !node->enforceBF16evenForGraphTail)
             continue;
 
-        if (one_of(node->getType(), Type::Input, Type::Output, Type::MemoryInput, Type::MemoryOutput))
+        // "Ignore the 'If' node because it contains two subgraphs. Enforcing the graph's input/output with the 'If'
+        // node will not be effective, and doing so will cause its output to differ from its subgraphs, resulting in
+        // memory out-of-bounds errors."
+        if (one_of(node->getType(), Type::Input, Type::Output, Type::MemoryInput, Type::MemoryOutput, Type::If))
```

Comment:
The comment isn't exactly the case, from the debug log we saw, **subgraphs in If node do enforced to bf16** (but for the if node in chatGLM, there were no computational intensive candidate nodes contained in the subgraph, thus none of the nodes in subgraphs is enforced to bf16)**

Whether any of nodes in subgraph is enforced to bf16 or not, the precision of input & output of the subgraph will be unchanged, so the input and output of If node itself should be also remained unchanged, since If node only copies I/O data from it's parent edges into subgraph's input or verse versa, it can not handle precision conversion.

Please update the comment, thanks!